semgrep/semgrep/rule.py

```diff
@@ -58,14 +61,14 @@ def __init__(self, raw: YamlTree[YamlMap]) -> None:
         self._excludes = path_dict.get("exclude", [])
         self._languages = [Language(l) for l in self._raw["languages"]]
 
-        # add 'ts' to languages if the rule supports javascript.
-        if ("javascript" in self._languages) or ("js" in self._languages):
-            self._languages.append(Language("ts"))
+        # add typescript to languages if the rule supports javascript.
+        if any(language in self._languages for language in JAVASCRIPT_LANGUAGES):
+            self._languages.extend(TYPESCRIPT_LANGUAGES)
```

Comment:
Per @aryx, adding all Typescript languages should be fine here because Typescript and TSX are considered equivalent :+1: